backend/mem/location.go

```diff
@@ -0,0 +1,211 @@
+}
+
+/*
+ListByRegex takes a regular expression and returns
+a slice of strings containing the base names of files
+found that matched the regular expression.  Returns an
+empty slice upon nothing found
+*/
+func (l *Location) ListByRegex(regex *regexp.Regexp) ([]string, error) {
+
+	list := make([]string, 0)
+	str := l.Path()
+	filesHere, _ := l.List()
+	for _, hereList := range filesHere {
+
+		potentialPath := path.Join(str, hereList)
+
+		for _, systemFileList := range fileList {
+
+			if systemFileList != nil && systemFileList.Path() == potentialPath {
+				ex, _ := systemFileList.Exists()
+				if regex.MatchString(path.Base(potentialPath)) && ex {
+					list = append(list, systemFileList.Name())
+
+				}
+
+			}
+		}
+
+	}
+
+	return list, nil
+}
+
+//Volume returns the volume of the current fs. In-Memory-Fs has no volume
+func (l *Location) Volume() string {
+	return ""
+}
+
+//Path returns the full, absolute path of the location with leading and trailing slashes
+func (l *Location) Path() string {
+
+	return l.name
+
+}
+
+//Exists reports whether or not a location exists. Creating a location does not guarantee its existence
+func (l *Location) Exists() (bool, error) {
+
+	data, _ := l.List()
+	if len(data) == 0 {
+		return false, nil
+	}
+	return true, nil
+
+}
+
+/*
+NewLocation creates a new location at the
+given relative path, which is tagged onto the current locations absolute path
+*/
+func (l *Location) NewLocation(relativePath string) (vfs.Location, error) {
+
+	str := path.Join(l.Path(), relativePath)
+	str = utils.AddTrailingSlash(path.Clean(str))
+	return &Location{
+		fileSystem: l.fileSystem,
+		name:       str,
+		exists:     false,
+	}, nil
+
+}
+
+//ChangeDir simply changes the directory of the location
+func (l *Location) ChangeDir(relativePath string) error {
+	l.name = path.Join(l.name, relativePath)
+	return nil
+
+}
+
+//FileSystem returns the type of filesystem location exists on, if it exists at all
+func (l *Location) FileSystem() vfs.FileSystem {
+
+	filePath := path.Join(path.Clean(l.Path()), l.Filename)
+	if systemMap[filePath] != nil {
+		if systemMap[filePath].exists {
+			l.exists = true
+		}
+	}
+	existence, _ := l.Exists()
+	if existence {
+		return l.fileSystem
+	}
+	return nil
+
+}
+
+//NewFile creates a vfs file given its relative path and tags it onto "l's" path
+func (l *Location) NewFile(fileName string) (vfs.File, error) {
+
+	pref := l.Path()
+	str := fileName
+	var nameStr string
+	if pref == "./" {
+		nameStr = path.Join("/", fileName)
+	} else {
+		nameStr = path.Join(pref, str)
+	}
+
+	loc, _ := l.fileSystem.NewLocation("", nameStr)
+	file := &File{timeStamp: time.Now(), isRef: false, Filename: path.Base(nameStr), cursor: 0,
+		isOpen: false, exists: false, location: loc}
+	systemMap[nameStr] = file
+	fileList = append(fileList, file)
+
+	return file, nil
+
+}
+
+//DeleteFile locates the file given the fileName and calls delete on it
+func (l *Location) DeleteFile(fileName string) error {
+
```

Comment:
Go's style is against whitespacing after function's signature.

Run this shell script and it will fix it everywhere (took me 10 minutes to write lol)

``` sh
sed -e '1h;2,$H;$!d;g' -e 's|{\n\s*|{|g;s|\s*\n}|}|g'
```

If you have sane GNU coreutils, run `find . -type f | grep '\.go$'` | xargs sed -e '1h;2,$H;$!d;g' -e 's|{\n\s*|{|g;s|\s*\n}|}|g' -i` and it will fix all of that formatting!

Reply:
I would avoid running this command against the entire project. Instead, ensure that your IDE is configured to automatically handle formatting.